micrometer-core/src/main/java/io/micrometer/core/instrument/binder/build/BuildInfoMetrics.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2023 VMware, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.micrometer.core.instrument.binder.build;
+
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.binder.MeterBinder;
+
+import java.time.Instant;
+
+import static java.util.Objects.requireNonNull;
+
+public class BuildInfoMetrics implements MeterBinder {
+
+    private final BuildInfo buildInfo;
+
+    public BuildInfoMetrics(BuildInfo buildInfo) {
+        requireNonNull(buildInfo, "buildInfo");
+        this.buildInfo = buildInfo;
+    }
+
+    @Override
+    public void bindTo(MeterRegistry registry) {
+        requireNonNull(registry, "registry");
+        Gauge.builder("build.info", () -> 1L)
+            .description("Build information")
+            .strongReference(true)
+            .tag("group", buildInfo.getGroup().orElse(DEFAULT_TAG_VALUE))
+            .tag("artifact", buildInfo.getArtifact().orElse(DEFAULT_TAG_VALUE))
+            .tag("name", buildInfo.getName().orElse(DEFAULT_TAG_VALUE))
+            .tag("version", buildInfo.getVersion().orElse(DEFAULT_TAG_VALUE))
+            .tag("timestamp", buildInfo.getTimestamp().map(Instant::toString).orElse(DEFAULT_TAG_VALUE))
```

Comment:
My initial thoughts on the `timestamp` was whether it is a good candidate for metrics. But, I guess it should be fine since it will most probably be tied to versions and timestamp is more of an informative tag than anything.

Reply:
But is this really for tracking build info of dependencies? or the service itself?

Reply:
I think it could be used for either the service and/or one or many dependencies. This implementation does not impose an opinion. Ultimately it was added to try and meet parity with the [SpringBoot](https://github.com/spring-projects/spring-boot/blob/dbe882aced8656b41d673b2542256c227bc528cd/spring-boot-project/spring-boot-tools/spring-boot-gradle-plugin/src/main/java/org/springframework/boot/gradle/tasks/buildinfo/BuildInfoProperties.java#L100) equivalent class.

One use case for a dependency would be to use this metric to assert that "all production services are using SpringBoot version 3.1.x or a version with a `timestamp` value from the last 3 months."

One could case for the service might be to use this metric in combination with something like "deployment.info" (which does not exist in Micrometer right now, but for the purposes of this discussion, lets say it captures at least a `timestamp` value of when the runtime was deployed). You could then use these two metrics together to understand how long it takes to get an application into production starting from the time the deployable artifact was created. This example is a bit more complicated and has it's flaws (such as rollbacks, rolling deployments, auto-scaling) but shared it as a potential idea.

Ultimately I agree with you, the timestamp value may end up being more informational than anything but I could imagine use cases for it to be useful and it meets SpringBoot feature parity..